src/tools/adig.c

```diff
@@ -243,14 +256,108 @@ static ares_bool_t read_cmdline(int argc, const char * const *argv,
     }
   }
 
-  config->args_processed = state.optind;
+  if (!is_rcfile) {
+    config->args_processed = state.optind;
+  }
   if (config->args_processed >= argc) {
     snprintf(config->error, sizeof(config->error), "missing query name");
     return ARES_FALSE;
   }
   return ARES_TRUE;
 }
 
+static ares_bool_t read_rcfile(adig_config_t *config)
+{
+  char         configdir[128];
+  unsigned int cdlen  = 0;
+  unsigned int cdsize = sizeof(configdir);
+
+#if !defined(WIN32)
+#  if !defined(__APPLE__)
+  char *configdir_xdg;
+#  endif
+  char *homedir;
+#endif
+
+  char          rcfile[128];
```

Comment:
might be better to use PATH_MAX instead of 128 here, likely need to include limits.h to get that

Reply:
For rcfile I replaced it as well now, but that yields format-truncation warning.